src/recipe.cpp

```diff
@@ -352,19 +348,28 @@ void recipe::load( const JsonObject &jo, const std::string &src )
 
 void recipe::finalize()
 {
-    if( test_mode && check_blueprint_needs ) {
+    if( bp_autocalc ) {
+        blueprint_reqs_ = get_build_reqs_for_furn_ter_ids(
+                              get_changed_ids_from_update( blueprint ) );
```

Comment:
The same here. `check_blueprint_requirements()` compares specified requirements with consolidated auto-calculated requirements, so `blueprint_reqs_` should be consolidated here too.

Reply:
`blueprint_reqs_` can't be consolidated; it's not a `requirement_data` object.  The auto-calculated requirements *are* consolidated inside `check_blueprint_requirements`, exactly the same as in your previous implementation, and to ensure consistency.